### ISSUE TITLE
Why plugin checks sha256 at all?

### ISSUE THREAD

**Issue by gavvvr** [franzbecker/gradle-lombok#9]
Hello. Maybe i am missing something, but why does this plugin check checksum and is bound to 1.16.4 version? What if i want to use latest 1.16.8 version and want to pass verifyLombok task, where should i get sha256 for another version?
I can skip specifying sha256, but verifyLombok will fail, because it will check against 1.16.4 hash.
I think comparing checksum is useless.

**Comment by franzbecker** [comment]
Hi there,

you are right that the default Lombok version of the current plugin is 1.16.4 and therefore the sha256 is configured to it. When using a newer version in combination with the `installLombok` you need to configure the expected sha256 as well. The sha256 of 1.16.8 should be `fe32b29b7b33eb2b19866cbc5345c59c9984a6036cc396856a04c0d09f99ea68`. Unfortunately, the project itself doesn't publish it, so I downloaded it from their website and from Maven central and compared both hashes. If you're on a Mac or Linux based system you can easily do that by executing `shasum -a 256 lombok.jar`.

I disagree with you that the comparison is useless. Comparing the checksum verifies the integrity of the downloaded JAR which gives you a chance to detect a manipulated JAR before installing it as your Java agent into Eclipse (which, I hope you agree, is not what you want). It is, of course, no guarantee as the JAR itself is not signed and the JAR that was used to calculate the hash could have already been manipulated. But in my opinion this is better than nothing. I tried building it from scratch but it yielded a different hash, which is often the case when using Maven (unlike e.g. Bazel). I could go on and on but I hope you get the general idea behind the `verifyLombok` task.

**Comment by gavvvr** [comment]
I got your point, thank you for explanation

**Comment by kpage** [comment]
Hello @franzbecker , thanks for this great plugin, it makes installation to the IDE easy for my team!  

I had one suggestion regarding the sha256 issue: gradle will automatically verify the checksum of an artifact retrieved from a maven repository.  So if your plugin is retrieving this artifact using standard dependency resolution, this check will be made for you automatically.  It might be easier for your plugin users to remove the manual check.  I also had to upgrade to 1.16.8 and I was not sure how to calculate the sha256 myself and from what artifact without reading the plugin source code.

If projects are concerned that the artifact might be compromised in maven central and thus the md5/sha1 files tampered with there, they can use something like https://github.com/whispersystems/gradle-witness for verification of all their dependencies.  But that level of verification might be better left to individual project users instead of being a requirement to use a particular gradle plugin.

**Comment by franzbecker** [comment]
Hello @kpage,

great to hear that the plugin helps your team :) 

It would be possible to disable the check but I argue that every project should be concerned that the artifact might be compromised before installing it as their Eclipse Java agent, especially when using Maven repositories without https.

Would it help if the plugin would print out more information and maybe a link to a wiki page that lists the current sha256 values?

**Comment by kpage** [comment]
Hi @franzbecker , you are right, it is a legitimate concern.  But Gradle will use https to access maven central by default if you declare mavenCentral() as a repository.  So your main concern would be that the artifact itself in maven central was tampered with.  But that's a general concern that individual gradle users could handle on all of their dependencies using gradle-witness as opposed to using different hash-checking schemes for each plugin.  Most projects would probably trust the security of maven central instead of going through this effort, because it does add additional effort.

If you would like to keep the sha256 check, I like your suggestion to publish the list of values and direct the user on how to calculate the sha256 themselves if new versions come out and the list gets stale.

**Comment by franzbecker** [comment]
Hi @kpage,

I wasn't worried so much about users that will access maven central by https but rather about those trapped behind a corporate proxy that breaks up the encryption and weakens your security.

But yes, I think your underlying point is valid. A project should be able to decide for itself how it wants to handle it, so I'm going to make the check optional. Hence, a project can easily use it but doesn't have to.

**Comment by sbraconnier** [comment]
First, thanks for this plugin! It helps a lot!!

Just my 2 cents on this. In our project we override the lombok version "blindly" by using the dependency-management-plugin provided by Spring:

```
....

// See https://github.com/spring-gradle-plugins/dependency-management-plugin
dependencyManagement {
    imports {
        mavenBom "org.springframework.boot:spring-boot-dependencies:1.3.5"
    }
}

....

lombok {
    version = dependencyManagement.importedProperties['lombok.version']
    sha256 = "..."
}
```

Since the lombok version is provided by spring-boot, we have to first run the lombok installation (installLombok) to see the expected sha256 value as an error message and put it back in the gradle script. In this particular case, making the check optional would be a big improvement for us.

Best regards!

**Comment by franzbecker** [comment]
I've just released version 1.7 where you can set `sha256` to null or an empty string and will thereby skip the `verifyLombok` task.

Thanks for the feedback, hope this helps :-)

**Comment by sbraconnier** [comment]
Wow, that was fast! It works like a charm! Many many thanks!
